kermit/src/commonMain/kotlin/co/touchlab/kermit/Logger.kt

```diff
@@ -10,26 +10,163 @@
 
 package co.touchlab.kermit
 
-abstract class Logger{
-    open fun isLoggable(severity: Severity): Boolean = true
+open class Logger(
+    val config: LoggerConfig,
+    val tag: String = config.defaultTag
+) {
+    fun withTag(tag: String): Logger {
+        return Logger(this.config, tag)
+    }
 
-    abstract fun log(severity: Severity, message: String, tag: String, throwable: Throwable? = null)
+    /**
+     * We could reduce the methods on this object, but the native objc
+     * export can't use default arguments, so we have a few extra methods defined.
+     */
```

Comment:
This will get interpreted by the IDE as doc comments on the function below it. I'd do a `/*` or `//`-style comment instead since it's more of an internal note to someone reading through the class.